Python/formatter_unicode.c

```diff
@@ -280,7 +281,10 @@ parse_internal_render_format_spec(PyObject *format_spec,
 
     if (end-pos > 1) {
         /* More than one char remain, invalid format specifier. */
-        PyErr_Format(PyExc_ValueError, "Invalid format specifier");
+        PyErr_Format(PyExc_ValueError,
+                     "Invalid format specifier '%s' for object of type '%.200s'",
+                     PyUnicode_AsUTF8(format_spec),
```

Comment:
I think you want to print out `format_spec` between positions `start` and `end`, since those are the only characters that are being parsed. That would fix the error message to only contain the format spec, not the entire part with braces in the `.format()` case. It might break some other tests, but I think it's the right thing to do.

For f-strings and the builtin `format()`, `start` is zero and `end` is the end of the format spec, so it won't affect that case.

Reply:
I think the error block would look something like:
```
    if (end-pos > 1) {
        /* More than one char remain, invalid format specifier. */

        /* Create a temporary object that contains the format spec we're
           operating on.  It's format_spec[start, end) (in half-open range
           nomenclature). */
        PyObject* actual_format_spec = PyUnicode_FromKindAndData(kind, (char*)data + kind*start, end-start);
        if (!actual_format_spec) {
            return 0;
        }

        PyErr_Format(PyExc_ValueError,
                     "Invalid format specifier '%s' for object of type '%.200s'",
                     PyUnicode_AsUTF8(actual_format_spec),
                     Py_TYPE(obj)->tp_name);

        Py_DECREF(actual_format_spec);
        return 0;
    }
```
This is ugly and I'm not sure the error handling is correct. I'm not sure of the best way to create a new PyUnicode object given an existing PyUnicode and a range. But it does produce the output I expect:
```
>>> format('', 'xxx')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid format specifier 'xxx' for object of type 'str'
>>> f'{"":xxx}'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid format specifier 'xxx' for object of type 'str'
>>> '{:xxx}'.format('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid format specifier 'xxx' for object of type 'str'
```

Reply:
I think this is a better version.
```
    if (end-pos > 1) {
        /* More than one char remains, so this is an invalid format
           specifier. */
        /* Create a temporary object that contains the format spec we're
           operating on.  It's format_spec[start:end] (in Python syntax). */
        PyObject* actual_format_spec = PyUnicode_FromKindAndData(kind,
                                         (char*)data + kind*start,
                                         end-start);
        if (actual_format_spec) {
            PyErr_Format(PyExc_ValueError,
                "Invalid format specifier '%U' for object of type '%.200s'",
                actual_format_spec, Py_TYPE(obj)->tp_name);
            Py_DECREF(actual_format_spec);
        }
        return 0;
    }
```